libzim/writer.py

```diff
@@ -70,6 +71,39 @@ def gen_blob(self):
         raise NotImplementedError("gen_blob (ro feed) must be implemented")
 
 
+class StringProvider(ContentProvider):
+    def __init__(self, content):
+        super().__init__()
+        self.content = content.encode("UTF-8") if isinstance(content, str) else content
+
+    def get_size(self):
+        return len(self.content)
+
+    def gen_blob(self):
+        yield Blob(self.content)
+
+
+class FileProvider(ContentProvider):
+    def __init__(self, filepath):
+        super().__init__()
+        self.size = os.path.getsize(filepath)
+        self.filehandler = open(filepath, "rb")
+
+    def get_size(self):
+        return self.size
+
+    def gen_blob(self):
+        bsize = 1048576  # 1MiB chunk
+        res = self.filehandler.read(bsize)
+        while res:
+            yield Blob(res)
+            res = self.filehandler.read(bsize)
```

Comment:
We can use a context manager here :
```
bsize = 1048576
with open(self.filepath, "rb") as f:
    res = f.read(bsize)
    while res:
        yield Blob(res)
        res = f.read(bsize) 
```
This way we don't have to store the filehandler (and close it in __del__).